### PR TITLE
Update Base Version in azure-pipelines.yml

### PR DESCRIPTION
**Pull request by twsl**
### Description of Change ###

 
This is a quick and dirty fix for the issue regarding the nightly versions by raising the semantic base version.

 ### Linked Issues ###
 

 - Fixes https://github.com/CommunityToolkit/Maui/issues/909

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [ ] Has tests (if omitted, state reason in description)
 - [ ] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [ ] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/XYZ 

 ### Additional information ###

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 variables:
-  CurrentSemanticVersionBase: '1.0.0'
+  CurrentSemanticVersionBase: '3.1.0'
```
Comment (VladislavAntonyuk) [inline azure-pipelines.yml:2]: should it be (current version + 1), so 3.2.0 at the moment of writing?

**Review by VladislavAntonyuk — DISMISSED** [review]

**Comment by brminnick** [comment]
Is there a way to automatically set its version number based on the version number of the latest release?

**Comment by bijington** [comment]
`git describe` might get you some of the way there...

https://git-scm.com/docs/git-describe

Sadly I don't know how we would determine the type of change and ultimately the change in version component. 

The best long term solution is to use a tool like GitVersion that just does it for us

**Comment by VladislavAntonyuk** [comment]
I think we can set the version number to 99.99.99.99, in that case it will always be a latest version, if user wants to use nightly builds

**Comment by brminnick** [comment]
> I think we can set the version number to 99.99.99.99

Works for me!

**Comment by twsl** [comment]
as proposed in the discussion, ideally something like [GitVersion](https://github.com/GitTools/GitVersion) would be used to automatically calculate the semver. Just using the same version number for everything might cause other problems in the future but could work for now

**Comment by VladislavAntonyuk** [comment]
as our next release is 4.0.0, let's for now use 4.1.0 and see how it works.
Ideally, we can use GitVersion as suggested.
I do not recommend use 99.0.0.00 in case somebody limits the latest version (>3.1.0 && <5.0.0)

**Comment by twsl** [comment]
Can someone else approve this so it gets merged?

**Comment by VladislavAntonyuk** [comment]
@brminnick, @jfversluis are you ok with the changes?

**Comment by brminnick** [comment]
If this is not urgent, I prefer that we implement a permanent solution that references the latest release number and works for both `CommunityToolkit.Maui` and `CommunityToolkit.Maui.MediaElement`. 

This is a pretty low-effort PR that just delays the problem and creates more work for us in the future; we'll have to open a PR to bump this version number after every release.

**Comment by VladislavAntonyuk** [comment]
@twsl Could you please investigate how we can use GitVersion solution?

**Comment by twsl** [comment]
I can try, the issue with that is that I cannot test the azure pipelines locally

**Comment by VladislavAntonyuk** [comment]
@twsl don't worry about it. If the build passes, you can find the version number in the GitHub actions result:
image

**Comment by twsl** [comment]
@bijington I'll give it a try now

**Comment by bijington** [comment]
> @bijington Feel free to tackle this. I did a quick test and just adding the nuget to the project causes issues and fails to build. Not sure if this requires https://github.com/GitTools/GitVersion/issues/3334. Adding the azure pipelines task could also be a solution tho

Thanks for having a look. I'll look into that issue and hope that we can workaround it

**Comment by twsl** [comment]
@dotnet-policy-service agree

**Comment by bijington** [comment]
@jfversluis this reminds me we need to have a chat about GitVersion

**Pull request closed without merging**